src/components/Icon/Icon.tsx

```diff
@@ -1,39 +1,60 @@
+import useTheme from 'hooks/useTheme';
 import * as React from 'react';
+import type { DivProps } from 'utils/common';
 
 import iconSelector from './assets/iconSelector';
-import { iconContainerStyle, iconStyle } from './Icon.style';
-import type { AcceptedIconNames } from './types';
-import type { colorShades } from '../../theme/palette';
-import type { AcceptedColorComponentTypes } from '../../utils/themeFunctions';
+import { iconContainerStyles, iconStyles } from './Icon.style';
+import type { AcceptedIconNames } from './Icon.types';
 import type { TestProps } from '../../utils/types';
 
-export type OwnProps = {
+export type IconProps = {
   /** This property defines witch icon to use */
   name: AcceptedIconNames;
-  /** Property indicating the color of the icon. Defaults to primary */
-  color?: AcceptedColorComponentTypes | string;
-  /** Property indicating the size of the icon. Defaults to 16 */
-  size?: number;
-  /** Callback fired when the `span` is clicked. */
-  onClick?: React.MouseEventHandler<HTMLSpanElement>;
+  /** Property indicating the color of the icon. Accepts all css:color values */
+  color?: string;
+  /**
+   * Property indicating the size of the icon. If number or string ending in 'px' is imported, it's converted to rem.
+   * Otherwise it's passed directly to the css:width/height. Defaults to 20px (converted to rem)
+   */
+  size?: string | number;
+  /** Whether the icon has a onHover style. Defaults to true if onClick is provided */
+  hasHover?: boolean;
+} & DivProps &
+  TestProps;
 
-  /** Property indicating the color's variant of the icon. */
-  variant?: (typeof colorShades)[number];
-};
+const Icon = React.forwardRef<HTMLDivElement, IconProps>((props, ref) => {
+  const theme = useTheme();
+  const {
+    name,
+    color = theme.tokens.colors.get('textColor.default.secondary'),
+    size = 20,
+    onClick,
+    dataTestId,
+    ...rest
+  } = props;
+  const isInteractive = Boolean(onClick);
+  const { hasHover = isInteractive } = props;
+  const Icon = iconSelector[name];
 
-export type IconProps = OwnProps & TestProps;
+  return (
+    <div
+      onClick={onClick}
+      css={iconContainerStyles({ size, hasHover, isInteractive })}
+      data-testid={dataTestId}
+      ref={ref}
+      tabIndex={isInteractive ? 0 : undefined}
+      onKeyDown={(e) => {
+        if (e.key === 'Enter' && onClick) {
+          onClick(null);
```

Comment:
i would recommend pass back the event
```suggestion
          onClick(e);
```